src/musica_aerosol_modal.F90

```diff
@@ -187,4 +191,25 @@ end subroutine sample_forward_scattering_optical_depth
 
 !!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
 
+  !> \section arg_table_aerosol_modal_init  Argument Table
+  !! \htmlinclude aerosol_modal_init.html
+  subroutine aerosol_modal_init(aerosol, errcode, errmsg)
+    type(aerosol_modal_t), intent(out) :: aerosol
+    integer,               intent(out) :: errcode
+    character(len=512),    intent(out) :: errmsg
+
+    errcode = 0
+    errmsg = ''
+  end subroutine aerosol_modal_init
+
+  !> \section arg_table_aerosol_modal_run  Argument Table
+  !! \htmlinclude aerosol_modal_run.html
+  subroutine aerosol_modal_run(errcode, errmsg)
+    integer,               intent(out) :: errcode
+    character(len=512),    intent(out) :: errmsg
+
+    errcode = 0
+    errmsg = ''
+  end subroutine aerosol_modal_run
+
```

Comment:
Is it required to include the `_init` and `_run` procedures even when they are not needed? Or is this just showing where they would be when they are needed?

Could the `_init` and `_run` procedures and accompanying metadata instead be added to the abstract `aerosol_t` class, and let extending classes override these as needed?

Reply:
`_init`, `_final`, `_timestep_init`, and `_timestep_final` are all optional.
`_run` might be optional, I've never tried (not normally how physics schemes are designed).

I do not understand the question about adding metadata to the abstract base class. CCPP scheme metadata is documentation of routines that are callable at the physics suite level. As such, they document the standard names (and other properties) for all of their arguments. Since an abstract type cannot be instantiated, there is no way to write such a routine so I do not see any benefit from writing metadata for it.

Reply:
I don't think you necessarily need to call the type-bound function, you could make wrapper functions if you want. But, the reason I was asking about moving the `_init` and `_run` functions to the abstract class (or the module its in, if you want to write wrapper functions), is that `_init` and `_run` would be functions common to all aerosol models - which is exactly what an abstract class is for.

Right now, the meta-data for radiation includes passing the standard-named variable `modal_aerosol_object` which works for this simple demo, but if we added another `aerosol_t` extending class for, say, carma, would we have to modify the meta-data every time we wanted to switch aerosol models? It would make much more sense to have a standard-name variable `aerosol_object` that would be polymorphic and could be mam, or carma, or anything else.